sdv/single_table/dayz.py

```diff
@@ -1,16 +1,232 @@
+
+
+def _validate_numerical_parameters(column_parameters, column_table_msg):
+    for param in ['min_value', 'max_value']:
+        if param in column_parameters and not _is_numerical(column_parameters[param]):
+            msg = f"'{param}' for {column_table_msg} must be a float."
+            raise SynthesizerProcessingError(msg)
+
+    if 'min_value' in column_parameters and 'max_value' in column_parameters:
+        if column_parameters['min_value'] > column_parameters['max_value']:
+            msg = (
+                f"Invalid parameters for {column_table_msg}. 'min_value' "
+                "must be less than 'max_value'"
+            )
+            raise SynthesizerProcessingError(msg)
+    elif 'min_value' in column_parameters or 'max_value' in column_parameters:
+        raise SynthesizerProcessingError(
+            f'Invalid parameters for {column_table_msg}. Both a '
+            "'min_value' and a 'max_value' must be set."
+        )
+
+
+def _validate_datetime_parameters(column_parameters, column_metadata, column_table_msg):
+    datetime_format = column_metadata.get('datetime_format')
+    for param in ['start_timestamp', 'end_timestamp']:
+        if param not in column_parameters:
+            continue
+
+        if param in column_parameters and not isinstance(column_parameters[param], str):
+            msg = f"'{param}' for {column_table_msg} must be a string."
+            raise SynthesizerProcessingError(msg)
+
+        if not _datetime_string_matches_format(column_parameters[param], datetime_format):
+            if datetime_format:
+                msg = (
+                    f"The '{param}' for {column_table_msg} is not a valid datetime string "
+                    f'or does not match the date time format ({datetime_format}).'
+                )
+            else:
+                msg = f"The '{param}' for {column_table_msg} is not a valid datetime string."
+
+        try:
+            pd.to_datetime(column_parameters[param], format=datetime_format)
+        except Exception as e:
+            raise SynthesizerProcessingError(msg) from e
+
+    if 'start_timestamp' in column_parameters and 'end_timestamp' in column_parameters:
+        start_datetime = pd.to_datetime(
+            column_parameters['start_timestamp'], format=datetime_format
+        )
+        end_datetime = pd.to_datetime(column_parameters['end_timestamp'], format=datetime_format)
+        if start_datetime > end_datetime:
+            msg = (
+                f"Invalid parameters for {column_table_msg}. The 'start_timestamp' must be "
+                "less than the 'end_timestamp'."
+            )
+            raise SynthesizerProcessingError(msg)
+    elif 'start_timestamp' in column_parameters or 'end_timestamp' in column_parameters:
+        raise SynthesizerProcessingError(
+            f'Invalid parameters for {column_table_msg}. Both a '
+            "'start_timestamp' and an 'end_timestamp' must be set."
+        )
```

Comment:
It's not possible to set only one of the two?

Reply:
Not according to the DayZ methods - it makes sense though since we shouldn't assume that the defaults are compatible with a provided value.